app/modules/assets/models.py

```diff
@@ -187,6 +187,8 @@ def get_or_make_master_format_path(self):
 
     def delete(self):
         with db.session.begin(subtransactions=True):
+            for annotation in self.annotations:
+                annotation.delete()
```

Comment:
We'll want to have the cascade tested at some point, but it shouldn't hold this up.